### DISCUSSION TITLE
Run the template analyzer in a YAML devops pipeline

### DISCUSSION THREAD

**Discussion by pim-simons** [Azure/template-analyzer#301] [Q&A] [answered]
First of all, great tool you are building! 👍🏻 

I am looking to include the template-analyzer as a proof of concept into my Azure YAML DevOps pipelines. I am currently using the ARM TTK tool but always looking to improve or extend the validation of my templates.
I know there is already a [Microsoft Security DevOps extension for Azure DevOps marketplace extension](https://github.com/microsoft/security-devops-azdevops) that includes the Template Analyzer, but that doesn't allow me to run the template-analyzer on specific templates or directories, also I can't really find if that extension runs the latest version of the template-analyzer or not.

Any guidance on how to run this from an Azure DevOps YAML pipeline? Is this already supported?
I know the tool is currently in development and not yet recommended for production usage, so I understand if this is not yet possible.

**Comment by VeraBE** [comment, marked as answer]
The Microsoft Security DevOps extension should allow you to change most of the TemplateAnalyzer settings. We are currently working on updating the official public documentation (https://learn.microsoft.com/en-us/azure/defender-for-cloud/azure-devops-extension), but until that's improved you can see the options here: https://github.com/microsoft/security-devops-action/wiki#templateanalyzer-options

We still don't have a strict release policy, but until now every time we release a new version of the tool we update it there, and we plan to do the same soon. The latest changes should be available there by around November 16

Besides that, we don't have plans to create an extension for this tool by itself

**Reply by pim-simons** [reply]
> The Microsoft Security DevOps extension should allow you to change most of the TemplateAnalyzer settings. We are currently working on updating the official public documentation ([learn.microsoft.com/en-us/azure/defender-for-cloud/azure-devops-extension](https://learn.microsoft.com/en-us/azure/defender-for-cloud/azure-devops-extension)), but until that's improved you can see the options here: [microsoft/security-devops-action/wiki#templateanalyzer-options](https://github.com/microsoft/security-devops-action/wiki#templateanalyzer-options)
> 
> We still don't have a strict release policy, but until now every time we release a new version of the tool we update it there, and we plan to do the same soon. The latest changes should be available there by around November 16
> 
> Besides that, we don't have plans to create an extension for this tool by itself

Thanks for [this link](https://github.com/microsoft/security-devops-action/wiki#templateanalyzer-options), that clears the config part up a bit.
Regarding the updates, is this only done for the [GitHub action](https://github.com/microsoft/security-devops-action) or also for the [DevOps extension](https://github.com/microsoft/security-devops-azdevops)?

**Reply by VeraBE** [reply]
> Thanks for [this link](https://github.com/microsoft/security-devops-action/wiki#templateanalyzer-options), that clears the config part up a bit. Regarding the updates, is this only done for the [GitHub action](https://github.com/microsoft/security-devops-action) or also for the [DevOps extension](https://github.com/microsoft/security-devops-azdevops)?

It's for both :)

**Reply by pim-simons** [reply]
Awesome, thanks!
